subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/TestDescriptor.java

```diff
@@ -41,6 +42,21 @@
     @Nullable
     String getClassName();
 
+    /**
+     * The name for display. It may be the same as or different from {@link #getName()}
+     *
+     * @return the name for display.
+     */
+    @Incubating
+    String getDisplayName();
+
+    /**
+     * The class name for display. It may be the same as or different from {@link #getClassName()}
+     * @return the class name for display.
+     */
+    @Incubating
+    String getClassDisplayName();
```

Comment:
Should we add tests that make sure this works in `beforeTest` and `afterTest` callbacks?